File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/properties/specs/FakeFilesPropertiesProvider.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.properties.specs;

import com.lasthopesoftware.bluewater.client.library.items.media.files.ServiceFile;
import com.lasthopesoftware.bluewater.client.library.items.media.files.properties.ProvideLibraryFileProperties;
import com.lasthopesoftware.bluewater.client.library.repository.LibraryId;
import com.namehillsoftware.handoff.promises.Promise;

import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;

public class FakeFilesPropertiesProvider implements ProvideLibraryFileProperties {
	private final Map<ServiceFile, Map<String, String>> cachedFileProperties = new HashMap<>();

	@NotNull
	@Override
	public Promise<Map<String, String>> promiseFileProperties(LibraryId libraryId, ServiceFile serviceFile) {
		try {
			return new Promise<>(cachedFileProperties.get(serviceFile));
		} catch (Throwable e) {
			return new Promise<>(e);
		}
	}

	public void addFilePropertiesToCache(ServiceFile serviceFile, Map<String, String> fileProperties) {
		cachedFileProperties.put(serviceFile, fileProperties);
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/properties/FakeFilesPropertiesProvider.kt
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.properties

import com.lasthopesoftware.bluewater.client.browsing.items.media.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.namehillsoftware.handoff.promises.Promise
import java.util.*

class FakeFilesPropertiesProvider : ProvideLibraryFileProperties {
    private val cachedFileProperties = HashMap<Pair<ServiceFile, LibraryId>, Map<String, String>>()
    override fun promiseFileProperties(libraryId: LibraryId, serviceFile: ServiceFile): Promise<Map<String, String>> {
        return try {
            Promise(cachedFileProperties[Pair(serviceFile, libraryId)])
        } catch (e: Throwable) {
            Promise(e)
        }
    }

    fun addFilePropertiesToCache(serviceFile: ServiceFile, libraryId: LibraryId, fileProperties: Map<String, String>) {
        cachedFileProperties[Pair(serviceFile, libraryId)] = fileProperties
    }
}
